aptos-move/framework/aptos-stdlib/sources/cryptography/groth16_bls12381_bellman.move

```diff
@@ -0,0 +1,70 @@
+module aptos_std::groth16_bls12381_bellman {
+
+    struct VerifyingKey has drop {
+        handle: u64,
+    }
+
+    struct Proof has drop {
+        handle: u64,
+    }
+
+    struct Scalar has drop {
+        handle: u64
+    }
+
+    public fun verify_proof(
+        _verifying_key: &VerifyingKey,
+        _public_inputs: &vector<Scalar>,
+        _proof: &Proof,
+    ): bool {
+        let public_input_handles: vector<u8> = vector[];
+        let num_public_inputs = std::vector::length(_public_inputs);
+        let i = 0;
+        while (i < num_public_inputs) {
+            std::vector::push_back(&mut public_input_handles, (std::vector::borrow(_public_inputs, i).handle as u8));
+            i = i + 1;
+        };
+
+        verify_proof_internal(_verifying_key.handle, public_input_handles, _proof.handle)
+    }
+
+    public fun new_verifying_key_from_bytes(bytes: vector<u8>): VerifyingKey {
+        VerifyingKey {
+            handle: new_verifying_key_from_bytes_internal(bytes)
+        }
+    }
+
+    public fun new_proof_from_bytes(bytes: vector<u8>): Proof {
+        Proof {
+            handle: new_proof_from_bytes_internal(bytes)
+        }
+    }
+
+    public fun new_scalar_from_bytes(bytes: vector<u8>): Scalar {
+        Scalar {
+            handle: new_scalar_from_bytes_internal(bytes)
+        }
+    }
+
+    native fun new_verifying_key_from_bytes_internal(bytes: vector<u8>): u64;
+    native fun new_proof_from_bytes_internal(bytes: vector<u8>): u64;
+    native fun new_scalar_from_bytes_internal(bytes: vector<u8>): u64;
+    native fun verify_proof_internal(vk_handle: u64, public_inputs: vector<u8>, proof_handle: u64): bool;
```

Comment:
Here, I understand you want a native because you are calling inside the `bellman` library.

To avoid the need for `get_scalar_handle` you can just use the same trick as in `ristretto255`'s `multi_scalar_mul_internal` native. Specifically, there I declare a native that takes a vector of points and scalars (not handles):

```
    /// The generic arguments are needed to deal with some Move VM peculiarities which prevent us from borrowing the
    /// points (or scalars) inside a &vector in Rust.
    ///
    /// WARNING: This function can only be called with P = RistrettoPoint and S = Scalar.
    native fun multi_scalar_mul_internal<P, S>(points: &vector<P>, scalars: &vector<S>): u64;
```

Then, in the native, in Rust, you can actually parse the handles (just see the Rust code for Ristretto).

Reply:
I guess your native would need to look like:

```
native verify_proof_internal<T>(vk_handle: u64, public_inputs: vector<Scalar<T>>, proof_handle: u64): bool;
```

This is slightly different than Ristretto, and we'd have to make sure things still work in Rust.

Reply:
Yes, the challenge will be to build a `Type` in Rust for `vector<Scalar<T>` given we already have a `Type` object for `T`.

We can coordinate with @vgao1996 on this, if we get confused.

Reply:
Oh, but actually, there's a simpler solution!